File: VB/ChangeDisplayNameForCalculatedField/Form1.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections
Imports System.ComponentModel
Imports System.ComponentModel.Design
Imports System.Windows.Forms
Imports DevExpress.XtraReports.UI
Imports DevExpress.XtraReports.UserDesigner
' ...

Namespace ChangeDisplayNameForCalculatedField
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub button1_Click(ByVal sender As Object, ByVal e As EventArgs) Handles button1.Click
			Dim rep As New XtraReport1()
			AddHandler rep.DesignerLoaded, AddressOf rep_DesignerLoaded
			rep.ShowDesignerDialog()
		End Sub

		Private Sub rep_DesignerLoaded(ByVal sender As Object, ByVal e As DesignerLoadedEventArgs)
			Dim svc As ITypeDescriptorFilterService = CType(e.DesignerHost.GetService(GetType(ITypeDescriptorFilterService)), ITypeDescriptorFilterService)
			Dim customSvc As New CustomTypeDescriptorFilterService(svc)
			e.DesignerHost.RemoveService(GetType(ITypeDescriptorFilterService))
			e.DesignerHost.AddService(GetType(ITypeDescriptorFilterService), customSvc)
		End Sub
	End Class

	Friend Class CustomTypeDescriptorFilterService
		Implements ITypeDescriptorFilterService
		Private svc As ITypeDescriptorFilterService
		Public Sub New(ByVal svc As ITypeDescriptorFilterService)
			Me.svc = svc
		End Sub

		#Region "ITypeDescriptorFilterService Members"
		Private Function FilterAttributes(ByVal component As IComponent, ByVal attributes As IDictionary) As Boolean Implements ITypeDescriptorFilterService.FilterAttributes
			Return svc.FilterAttributes(component, attributes)
		End Function
		Private Function FilterEvents(ByVal component As IComponent, ByVal events As IDictionary) As Boolean Implements ITypeDescriptorFilterService.FilterEvents
			Return svc.FilterEvents(component, events)
		End Function
		Private Function FilterProperties(ByVal component As IComponent, ByVal properties As IDictionary) As Boolean Implements ITypeDescriptorFilterService.FilterProperties
			Dim result As Boolean = svc.FilterProperties(component, properties)
			If TypeOf component Is CalculatedField Then
				Dim attributes() As Attribute = { BrowsableAttribute.Yes }
				properties("DisplayName") = TypeDescriptor.CreateProperty(GetType(CalculatedField), CType(properties("DisplayName"), PropertyDescriptor), attributes)
			End If
			Return result
		End Function
		#End Region
	End Class
End Namespace

File: CS/ChangeDisplayNameForCalculatedField/Form1.cs
using System;
using System.Collections;
using System.ComponentModel;
using System.ComponentModel.Design;
using System.Windows.Forms;
using DevExpress.XtraReports.UI;
using DevExpress.XtraReports.UserDesigner;
// ...

namespace ChangeDisplayNameForCalculatedField {
    public partial class Form1 : Form {
        public Form1() {
            InitializeComponent();
        }

        private void button1_Click(object sender, EventArgs e) {
            XtraReport1 rep = new XtraReport1();
            rep.DesignerLoaded += new DesignerLoadedEventHandler(rep_DesignerLoaded);
            rep.ShowDesignerDialog();
        }

        void rep_DesignerLoaded(object sender, DesignerLoadedEventArgs e) {
            ITypeDescriptorFilterService svc = (ITypeDescriptorFilterService)e.DesignerHost.GetService(typeof(ITypeDescriptorFilterService));
            CustomTypeDescriptorFilterService customSvc = new CustomTypeDescriptorFilterService(svc);
            e.DesignerHost.RemoveService(typeof(ITypeDescriptorFilterService));
            e.DesignerHost.AddService(typeof(ITypeDescriptorFilterService), customSvc);
        }
    }

    class CustomTypeDescriptorFilterService : ITypeDescriptorFilterService {
        ITypeDescriptorFilterService svc;
        public CustomTypeDescriptorFilterService(ITypeDescriptorFilterService svc) {
            this.svc = svc;
        }

        #region ITypeDescriptorFilterService Members
        bool ITypeDescriptorFilterService.FilterAttributes(IComponent component, IDictionary attributes) {
            return svc.FilterAttributes(component, attributes);
        }
        bool ITypeDescriptorFilterService.FilterEvents(IComponent component, IDictionary events) {
            return svc.FilterEvents(component, events);
        }
        bool ITypeDescriptorFilterService.FilterProperties(IComponent component, IDictionary properties) {
            bool result = svc.FilterProperties(component, properties);
            if (component is CalculatedField) {
                Attribute[] attributes = new Attribute[] { BrowsableAttribute.Yes };
                properties["DisplayName"] = TypeDescriptor.CreateProperty(typeof(CalculatedField), (PropertyDescriptor)properties["DisplayName"], attributes);
            }
            return result;
        }
        #endregion
    }
}
